osu.Game.Rulesets.Taiko/Mods/TaikoModClassic.cs

```diff
@@ -3,21 +3,200 @@
+
+            var hitObjects = taikoBeatmap.HitObjects.Select(ho =>
+            {
+                if (ho is DrumRoll drumRoll)
+                {
+                    var newDrumRoll = new ClassicDrumRoll(drumRoll);
+                    return newDrumRoll;
+                }
+
+                if (ho is Swell swell)
+                {
+                    var newSwell = new ClassicSwell(swell);
+                    return newSwell;
+                }
+
+                return ho;
+            }).ToList();
+
+            taikoBeatmap.HitObjects = hitObjects;
+        }
+
+        private class ClassicDrumRoll : DrumRoll
+        {
+            public ClassicDrumRoll(DrumRoll original)
+            {
+                StartTime = original.StartTime;
+                Samples = original.Samples;
+                EndTime = original.EndTime;
+                Duration = original.Duration;
+                TickRate = original.TickRate;
+                RequiredGoodHits = original.RequiredGoodHits;
+                RequiredGreatHits = original.RequiredGreatHits;
+            }
+
+            public override Judgement CreateJudgement() => new TaikoClassicDrumRollJudgement();
+
+            protected override void CreateTicks(CancellationToken cancellationToken)
+            {
+                if (TickSpacing == 0)
+                    return;
+
+                bool first = true;
+
+                for (double t = StartTime; t < EndTime + TickSpacing / 2; t += TickSpacing)
+                {
+                    cancellationToken.ThrowIfCancellationRequested();
+
+                    AddNested(new ClassicDrumRollTick
+                    {
+                        FirstTick = first,
+                        TickSpacing = TickSpacing,
+                        StartTime = t,
+                        IsStrong = IsStrong
+                    });
+
+                    first = false;
+                }
+            }
+        }
+
+        private class ClassicDrumRollTick : DrumRollTick
+        {
+            public override Judgement CreateJudgement() => new TaikoClassicDrumRollTickJudgement();
+        }
+
+        private class ClassicSwell : Swell
+        {
+            public ClassicSwell(Swell original)
+            {
+                StartTime = original.StartTime;
+                Samples = original.Samples;
+                EndTime = original.EndTime;
+                Duration = original.Duration;
+                RequiredHits = original.RequiredHits;
+            }
+
+            public override Judgement CreateJudgement() => new TaikoClassicSwellJudgement();
+        }
+
+        private class TaikoClassicDrumRollJudgement : TaikoDrumRollJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
+        }
+
+        private class TaikoClassicDrumRollTickJudgement : TaikoDrumRollTickJudgement
+        {
+            public override HitResult MaxResult => HitResult.LargeBonus;
```

Comment:
Why are ticks worth the same as the drum roll itself? I think `SmallBonus` would make more sense. Or make the drum roll itself not worth anything (`IgnoreHit`) and only award points for the drum roll.